app/packages/looker/src/worker/painter.ts

```diff
@@ -328,6 +331,18 @@ export const PainterFactory = (requestColor) => ({
           ) {
             targets[i] = 0;
           } else {
+            if (coloring.by == COLOR_BY.VALUE) {
+              // Attempt to find a color in the fields mask target color settings
+              // If not found, attempt to find a color in the default mask target colors.
+              const customColor =
+                fieldSetting?.[targets[i].toString()] ??
+                defaultSetting?.[targets[i].toString()];
+
+              // If a customized color setting is found, get the 32-bit color representation.
+              if (customColor) {
+                color = get32BitColor(convertToHex(customColor));
+              }
+            }
             overlay[i] = color ? color : getColor(targets[i]);
           }
         }
```

Comment:
@sashankaryal @benjaminpkane @manivoxel51 
let me know if you think this for loop can be further optimized for performance.

Reply:
Will follow up on this after Thanksgiving! Maybe we can improve a little